tree/ntuple/v7/inc/ROOT/RNTupleIndex.hxx

```diff
@@ -176,14 +228,14 @@ public:
    ///
    /// \return The entry numbers that corresponds to `valuePtrs`. When no such entry exists, an empty vector is
    /// returned.
-   const std::vector<NTupleSize_t> *GetAllEntryNumbers(const std::vector<void *> &valuePtrs) const;
+   const std::vector<NTupleSize_t> GetAllEntryNumbers(const std::vector<void *> &valuePtrs) const;
```

Comment:
Note that this interface forces the allocation of a `std::vector` including its heap-backed data for every probe

Reply:
Additionally, returning a `const` value is generally not what you want (as it is my understanding it may prevent move semantics/rvalue reference semantics, and it doesn't really give any benefit to the caller).
To avoid forcing the allocation on the caller while giving them ownership of the data it's probably better to pass the out vector as a reference parameter (so they may reuse it across calls etc).